extensions/spring-xd-extension-gpfdist/src/main/java/org/springframework/xd/greenplum/gpfdist/GPFDistMessageHandler.java

```diff
@@ -127,49 +134,93 @@ protected void doStart() {
 			context.addLocation(NetworkUtils.getGPFDistUri(gpfdistServer.getLocalPort()));
 
 			sqlTaskScheduler.schedule((new FutureTask<Void>(new Runnable() {
+
 				@Override
 				public void run() {
 					boolean taskValue = true;
 					try {
-						while(!taskFuture.interrupted) {
+						while (!taskFuture.interrupted) {
 							try {
 								greenplumLoad.load(context);
-							} catch (Exception e) {
+							}
+							catch (Exception e) {
 								log.error("Error in load", e);
 							}
-							Thread.sleep(batchPeriod*1000);
+							Thread.sleep(batchPeriod * 1000);
 						}
-					} catch (Exception e) {
+					}
+					catch (Exception e) {
 						taskValue = false;
 					}
 					taskFuture.set(taskValue);
 				}
 			}, null)), new Date());
 
-		} else {
+		}
+		else {
 			log.info("Skipping gpload tasks because greenplumLoad is not set");
 		}
 	}
 
 	@Override
 	protected void doStop() {
+		boolean drained = false;
 		if (greenplumLoad != null) {
+
+			// xd waits 30s to shutdown module, so lets wait 25 to drain
```

Comment:
Can you explain this a bit more - sounds like some sort of race condition, shouldn't we let the entire buffer drain since the messages in the buffer have been ack'd (say in rabbit) ?